### PR TITLE
Removed memory leak in function mongoc_database_has_collection

### PR DESCRIPTION
**Pull request by canella**
If an invalid namespace is supplied to mongoc_database_has_collection the cursor that is used in this function is not freed causing a memory leak every time the function is called. This leads to a drastic increase in memory consumption if the function is called frequently.

### CODE REVIEW

**Comment by canella** [comment]
Two pictures showing the memory consumption before and after applying the fix. In both cases the program ran over a minute while calling the function continuously.

mongoc_database_has_collection_fix
mongoc_database_has_collection_nofix

File: src/mongoc/mongoc-database.c [hunk 1]
```diff
@@ -618,6 +618,7 @@ mongoc_database_has_collection (mongoc_database_t *database,
        (error &&
         ((error->domain != 0) ||
          (error->code != 0)))) {
+      mongoc_cursor_destroy( cursor );
```
Comment (bjori) [inline src/mongoc/mongoc-database.c]: If we enter this clause with !cursor then mongoc_cursor_destroy() is going to be very pissed off.

Comment (canella) [inline src/mongoc/mongoc-database.c]: You are of course right, I had only tested it with an error, not an invalid cursor. I will fix this tonight by separating the if clause for cursor and error.

**Comment by ajdavis** [comment]
Thanks for the contribution!

**Pull request merged**